test/GPv2AccessControl.test.ts

```diff
@@ -0,0 +1,94 @@
+import { expect } from "chai";
+import { Contract } from "ethers";
+import { ethers, waffle } from "hardhat";
+
+describe("GPv2SimpleAuthentication", () => {
+  const [owner, nonOwner, solver] = waffle.provider.getWallets();
+  let authenticator: Contract;
+
+  beforeEach(async () => {
+    const GPv2SimpleAuthentication = await ethers.getContractFactory(
+      "GPv2SimpleAuthentication",
+    );
+
+    // Owner will default to admin account declared above.
+    authenticator = await GPv2SimpleAuthentication.deploy();
+  });
+
+  describe("constructor", () => {
+    it("should set deployer as owner", async () => {
+      expect(await authenticator.owner()).to.equal(owner.address);
+    });
+  });
+
+  describe("addSolver", () => {
+    it("should allow to add solver", async () => {
+      await expect(authenticator.addSolver(solver.address)).to.not.be.reverted;
+    });
+
+    it("should not allow non-owner to add solver", async () => {
+      await expect(
+        authenticator.connect(nonOwner).addSolver(solver.address),
+      ).to.be.revertedWith("caller is not the owner");
+    });
+  });
+
+  describe("removeSolver", () => {
+    it("should allow owner to add solver", async () => {
```

Comment:
<a href="https://gitme.me/image?url=https%3A%2F%2Ffirebasestorage.googleapis.com%2Fv0%2Fb%2Fgit-meme-prod.appspot.com%2Fo%2Fimages%252fcopypasta.jpg%3Falt%3Dmedia%26token%3De1497fbc-8c13-4e8e-b86d-22b0ae5976ee&token=copypasta" data-gitmeme-token="copypasta"><img src="https://firebasestorage.googleapis.com/v0/b/git-meme-prod.appspot.com/o/images%2fcopypasta.jpg?alt=media&token=e1497fbc-8c13-4e8e-b86d-22b0ae5976ee" title="Created by gitme.me with /copypasta"/></a>

Should this be `remove`?

Reply:
```suggestion
    it("should allow owner to remove solver", async () => {
```